codeQuality/index.md

```diff
@@ -0,0 +1,173 @@
+# Code Quality
+
+## Linting / Prettier
+
+In the CDapp we have linting rules enabled. When commiting, a number of pre-commit checks will run to ensure your code is consistent with these.
+
+Many of us also use Prettier to ensure that our code is formatted neatly on the page
+
+## Refactoring
+
+Set aside time / issues for refactoring… but not too early (nor too late). Generally when you start to notice copy/pasted code or that files are becoming too long, it's a good sign that you could start refactoring.
+
+## Types
+
+We generally try to be as specific as possible with our types. That means avoiding `any` and type-casting.
+
+Whenever we create a component with props, we always create an `interface` to describe the shape of the props object. If these props are being passed through multiple components, please reuse the interfaces higher up either by importing them as is, or by extending them in the child component.
+
+### Defining types in components
+
+If you're defining a type inside of a component file, ensure you define it at the top of the file and outside of the component.
+
+## React best practices
+
+We generally follow guidance from the [new react docs](https://react.dev/) when assessing react-specific best practices.
+
+Answers to the following questions and more can be found therein:
+
+- When to use useMemo/useCallback (hint: not unless you have to)
+- When (and when not!) to use useEffect
+- How to correctly implement useContext
+
+If a question can't be resolved by the react docs (or other authoritative source), then we default to sound programming practices (DRY/SOLID etc.). Finally, if neither of the previous stages produce a definitive answer, the preference of the person committing takes precedence.
+
+With that said, here are some other things we like to see.
+
+### Using useContext
+
+We generally prefer to use `useContext` than prop drill to pass state down multiple levels of the component hierarchy.
```

Comment:
I'd expand this to emphasize that we prefer this pattern for most data acquisition inside components.

And if devs run into parts that don't use it, they should strive to refactor the logic into a similar hook.